libcxx/include/__expected/expected.h

```diff
@@ -88,8 +88,263 @@ _LIBCPP_HIDE_FROM_ABI void __throw_bad_expected_access(_Arg&& __arg) {
 #  endif
 }
 
```

Comment:
Not attached to this line but I'd like to mention here. We have the exact same problems in `std::ranges::__movable_box`, which affects lots of views. Whatever the fix here is likely needed to fix `std::ranges::__movable_box`. So while you are fixing this, perhaps consider put some reusable logic separate so that we can reuse it later.